conda/core/solve.py

```diff
@@ -626,6 +626,8 @@ def _add_specs(self, ssc):
                     python_spec = MatchSpec(python_spec, version=pinned_version)
 
                 spec_set = (python_spec, ) + tuple(self.specs_to_add)
+                if len(conflict_specs) > 0 and self._repodata_fn == "current_repodata.json":
```

Comment:
More general to say "only find conflicts when we're using repodata.json, as it is the ultimate fallback.

```suggestion
                if len(conflict_specs) > 0 and self._repodata_fn != "repodata.json":
```